internal/pkg/config/workload.go

```diff
@@ -0,0 +1,243 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package config
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/ssm"
+)
+
+const (
+	// ServiceCfnTemplateNameFormat is the base output file name when `service package`
+	// is called. This is also used to render the pipeline CFN template.
+	ServiceCfnTemplateNameFormat = "%s.stack.yml"
+	// ServiceCfnTemplateConfigurationNameFormat is the base output configuration
+	// file name when `service package` is called. It's also used to render the
+	// pipeline CFN template.
+	ServiceCfnTemplateConfigurationNameFormat = "%s-%s.params.json"
+	// AddonsCfnTemplateNameFormat is the addons output file name when `service package`
+	// is called.
+	AddonsCfnTemplateNameFormat = "%s.addons.stack.yml"
+
+	serviceString = "service"
+	jobString     = "job"
+)
+
+// Workload represents a deployable long running service or task.
+type Workload struct {
+	App  string `json:"app"`  // Name of the app this workload belongs to.
+	Name string `json:"name"` // Name of the workload, which must be unique within a app.
+	Type string `json:"type"` // Type of the workload (ex: Load Balanced Web Service, etc)
+}
+
+// CreateService instantiates a new service within an existing application. Skip if
+// the service already exists in the application.
+func (s *Store) CreateService(svc *Workload) error {
+	if err := s.createWorkload(svc, serviceString); err != nil {
+		return fmt.Errorf("create service %s in application %s: %w", svc.Name, svc.App, err)
+	}
+	return nil
+}
+
+// CreateJob instantiates a new job within an existing application. Skip if the job already
+// exists in the application.
+func (s *Store) CreateJob(job *Workload) error {
+	if err := s.createWorkload(job, jobString); err != nil {
+		return fmt.Errorf("create job %s in application %s: %w", job.Name, job.App, err)
+	}
+	return nil
+}
+
+func (s *Store) createWorkload(wkld *Workload, wkldType string) error {
+	if _, err := s.GetApplication(wkld.App); err != nil {
+		return err
+	}
+
+	wkldPath := fmt.Sprintf(fmtWkldParamPath, wkld.App, wkld.Name)
+	data, err := marshal(wkld)
+	if err != nil {
+		return fmt.Errorf("serialize data: %w", err)
+	}
+
+	_, err = s.ssmClient.PutParameter(&ssm.PutParameterInput{
+		Name:        aws.String(wkldPath),
+		Description: aws.String(fmt.Sprintf("Copilot %s %s", strings.Title(wkldType), wkld.Name)),
```

Comment:
Should we remove the extra `wkldType` param to the method and instead have this description just be:
```suggestion
		Description: aws.String(fmt.Sprintf("Copilot %s %s", wlkd.Type, wkld.Name)),
```